fuse/dl/lightning/pl_module.py

```diff
@@ -160,3 +165,112 @@ def configure_optimizers(self) -> torch.optim.Optimizer:
     def set_predictions_keys(self, keys: List[str]) -> None:
         """Define which keys to extract from batch_dict on prediction mode"""
         self._prediction_keys = keys
+
+
+class BalancedLightningDataModule(pl.LightningDataModule):
```

Comment:
Not sure about the naming :)) 

Plus, it might be a better design choice to supply the custom batch sampler instead of raising the flag (`use_custom_batch_sampler`, see below)

Reply:
@mosheraboh 

In second thought, maybe it will be better to delete this Module and just show a usage of LightningDataModule.